facade/dfs.go

```diff
@@ -605,8 +605,15 @@ func (f *Facade) Restore(ctx datastore.Context, r io.Reader, backupInfo *dfs.Bac
 		if err := f.Rollback(ctx, snapshot, false); err != nil {
 			logger.WithError(err).Debug("Could not rollback snapshot")
 			return alog.Error(err)
+		} else {
+			logger.Info("Rolled back snapshot")
+			if err := f.dfs.Delete(snapshot); err != nil {
+				logger.WithError(err).Error("Problem deleting snapshot.")
+			} else {
+				logger.Info("Removed snapshot after rollback")
```

Comment:
Wouldn't it be better to untag when a snapshot deletion fails so that it can be removed automatically due to TTL?: https://github.com/control-center/serviced/blob/5968d0ec138bae02cc177bf3d5d3f81a9b584104/facade/dfs.go#L115

Reply:
Young's suggestion looks right.  If the deletion fails, in addition to the log entry we should untag it so that the ttl code can try to remove it.